src/test/libcephfs/snapdiff.cc

```diff
@@ -0,0 +1,1669 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+/*
+ * Ceph - scalable distributed file system
+ *
+ *
+ * This is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License version 2.1, as published by the Free Software
+ * Foundation.  See file COPYING.
+ *
+ */
+
+#include "gtest/gtest.h"
+#include "include/cephfs/libcephfs.h"
```

Comment:
The Windows job fails with:

```
In file included from /home/ubuntu/ceph/src/test/libcephfs/snapdiff.cc:15:
/home/ubuntu/ceph/src/include/cephfs/libcephfs.h:161:29: error: ‘uid_t’ was not declared in this scope; did you mean ‘pid_t’?
  161 | UserPerm *ceph_userperm_new(uid_t uid, gid_t gid, int ngids, gid_t *gidlist);
      |                             ^~~~~
      |                             pid_t
```

The reason is that uid_t is not available on Windows, being defined by ``src/include/compat.h``. Unfortunately we can't include ``compat.h`` in ``libcephfs.h``, so we can either include ``compat.h`` here or move the ``uid_t`` and ``gid_t`` definitions to ``src/include/win32/fs_compat.h``.